Experimental environment for emulating a device shell

estenberg opened the pull request:
Confirmed to work for doing `apt upgrade` with Raspberry Pi 3.

estenberg commented:
If this is generally useful and works we can generalize it (remove the mender-artifact generation) and put it in its own repo.

apodogrocki commented on device-image-shell/docker-device-image-shell:
```diff
@@ -0,0 +1,58 @@
+echo "Image generation complete!"
+echo "The new root file system is at:\n\t$OUTPUT_DIR/$ROOTFS_OUTPUT_FILE_NAME"
+echo "The new Mender Artifact you can upload to your Mender server to deploy to your devices is at:\
+         \n\t$OUTPUT_DIR/$ARTIFACT_NAME.mender"
```
`echo -e` to enable interpretation of backslash escapes

estenberg replied:
Thanks! Replaced with echo -e.

estenberg commented on device-image-shell/docker-device-image-shell:
```diff
@@ -0,0 +1,58 @@
+CURRENT_GROUP=$(id -g -n)
+
+echo "Changing ownership of Mender Artifact (may ask you to authenticate)"
+sudo chown $CURRENT_USER:$CURRENT_GROUP $OUTPUT_DIR/$ARTIFACT_NAME.mender
```
Thanks, though the chown did not have needed perms without sudo for me.
The Mender Artifact is owned by root (it is created inside container, which is run privileged).

If you have ideas I would be happy to get rid of this.

estenberg commented:
Thanks for testing Mirza:
* Added echo -e
* Added note that commands are run as root

It makes sense to run "mount -t devpts devpts /dev/pts", however I am not sure how to run commands after it enters chroot automatically - it is up to the user then?
